### PR TITLE
brave_stats: Add walletActive parameter

### PR DESCRIPTION
**Pull request by keur**
Resolves https://github.com/brave/brave-browser/issues/17059

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
In linked issue

### CODE REVIEW

**Comment by diracdeltas** [comment]
security review passed (https://github.com/brave/security/issues/462). please make sure https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Capture-of-Usage-Data gets updated once this is merged.

File: browser/brave_stats/brave_stats_updater_params.cc
Comment (bbondy) [inline browser/brave_stats/brave_stats_updater_params.cc:107]: What stops this from reporting a daily active user twice if the browser is restarted?  Could you also add a test case for that, that a single user can't self report twice in a row on the same day?

Comment (keur) [reply browser/brave_stats/brave_stats_updater_params.cc:107]: None of that logic changed in this PR. This just adds a param based off the timestamp you created. We already guard against that for sending the ping at all here https://github.com/brave/brave-core/blob/c9ba8ea1582145e11b26d849135094ffd8c2c257/browser/brave_stats/brave_stats_updater.cc#L288-L293

We can create that test case as a followup PR.

**Review by bbondy — APPROVED** [review]
ok sounds good

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.30.41 Chromium: 93.0.4577.51 (Official Build) nightly (64-bit)
-- | --
Revision | 762d21050e2da59930c784c09b134d0b0b148188-refs/branch-heads/4577@{#915}
OS | Windows 10 OS Version 2009 (Build 19042.1165)
```

#### `Scenario #1`

Launching a new profile, ensured that `walletActive=0` as per the following:

* `/1/usage/brave-core?platform=winx64-bc&channel=nightly&version=1.30.41&daily=true&weekly=true&monthly=true&first=true&woi=2021-08-23&dtoi=2021-08-23&ref=BRV001&adsEnabled=false&arch=&walletActive=0`

Closed the browser/moved the current date/time ahead by one day (2021-08-23 -> 2021-08-24) and ensured that `walletActive=0` as per the following:

* `/1/usage/brave-core?platform=winx64-bc&channel=nightly&version=1.30.41&daily=true&weekly=false&monthly=false&first=false&woi=2021-08-23&dtoi=2021-08-23&ref=BRV001&adsEnabled=false&arch=&walletActive=0`

#### `Scenario #2`

**`Step #1 - Daily, weekly, monthly`**

* `/1/usage/brave-core?platform=winx64-bc&channel=nightly&version=1.30.41&daily=true&weekly=false&monthly=false&first=false&woi=2021-08-23&dtoi=2021-08-23&ref=BRV001&adsEnabled=false&arch=&walletActive=7`

**`Step #2 - Weekly, monthly`**

* `https://laptop-updates.brave.com/1/usage/brave-core?platform=winx64-bc&channel=nightly&version=1.30.41&daily=true&weekly=false&monthly=false&first=false&woi=2021-08-23&dtoi=2021-08-23&ref=BRV001&adsEnabled=false&arch=&walletActive=6`

**`Step #3 - Monthly`**

* `https://laptop-updates.brave.com/1/usage/brave-core?platform=winx64-bc&channel=nightly&version=1.30.41&daily=true&weekly=true&monthly=false&first=false&woi=2021-08-23&dtoi=2021-08-23&ref=BRV001&adsEnabled=false&arch=&walletActive=4`

**`Step #4 - Re-enabling`**

* `https://laptop-updates.brave.com/1/usage/brave-core?platform=winx64-bc&channel=nightly&version=1.30.41&daily=true&weekly=false&monthly=true&first=false&woi=2021-08-23&dtoi=2021-08-23&ref=BRV001&adsEnabled=false&arch=&walletActive=7`

**Pull request merged**
